packages/contracts-bedrock/test/CommonTest.t.sol

```diff
@@ -31,46 +37,186 @@ import { AddressManager } from "src/legacy/AddressManager.sol";
+    L1ERC721Bridge l1ERC721Bridge;
+    OptimismMintableERC20Factory l1OptimismMintableERC20Factory;
+    ProtocolVersions protocolVersions;
+
+    L2CrossDomainMessenger l2CrossDomainMessenger;
+    L2StandardBridge l2StandardBridge;
+    L2ToL1MessagePasser l2ToL1MessagePasser;
+    OptimismMintableERC20Factory l2OptimismMintableERC20Factory;
+    L2ERC721Bridge l2ERC721Bridge;
+    BaseFeeVault baseFeeVault;
+    SequencerFeeVault sequencerFeeVault;
+    L1FeeVault l1FeeVault;
+    GasPriceOracle gasPriceOracle;
+    L1Block l1Block;
+    LegacyMessagePasser legacyMessagePasser;
+    GovernanceToken governanceToken;
+
     FFIInterface ffi;
 
-    function setUp() public virtual {
+    function setUp() public virtual override {
         // Give alice and bob some ETH
         vm.deal(alice, 1 << 16);
         vm.deal(bob, 1 << 16);
-        vm.deal(multisig, 1 << 16);
 
         vm.label(alice, "alice");
         vm.label(bob, "bob");
-        vm.label(multisig, "multisig");
 
         // Make sure we have a non-zero base fee
         vm.fee(1000000000);
 
+        // Set the deterministic deployer in state
+        vm.etch(
+            0x4e59b44847b379578588920cA78FbF26c0B4956C,
+            hex"7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3"
+        );
+
         ffi = new FFIInterface();
+
+        Deploy.setUp();
+        Deploy.run();
+
+        // Set up L1
+        optimismPortal = OptimismPortal(mustGetAddress("OptimismPortalProxy"));
+        l2OutputOracle = L2OutputOracle(mustGetAddress("L2OutputOracleProxy"));
+        systemConfig = SystemConfig(mustGetAddress("SystemConfigProxy"));
+        l1StandardBridge = L1StandardBridge(mustGetAddress("L1StandardBridgeProxy"));
+        l1CrossDomainMessenger = L1CrossDomainMessenger(mustGetAddress("L1CrossDomainMessengerProxy"));
+        addressManager = AddressManager(mustGetAddress("AddressManager"));
+        l1ERC721Bridge = L1ERC721Bridge(mustGetAddress("L1ERC721BridgeProxy"));
+        l1OptimismMintableERC20Factory =
+            OptimismMintableERC20Factory(mustGetAddress("OptimismMintableERC20FactoryProxy"));
+        protocolVersions = ProtocolVersions(mustGetAddress("ProtocolVersionsProxy"));
+
+        vm.label(address(l2OutputOracle), "L2OutputOracle");
+        vm.label(address(optimismPortal), "OptimismPortal");
+        vm.label(address(systemConfig), "SystemConfig");
+        vm.label(address(l1StandardBridge), "L1StandardBridge");
+        vm.label(address(l1CrossDomainMessenger), "L1CrossDomainMessenger");
+        vm.label(address(addressManager), "AddressManager");
+        vm.label(address(l1ERC721Bridge), "L1ERC721Bridge");
+        vm.label(address(l1OptimismMintableERC20Factory), "OptimismMintableERC20Factory");
+        vm.label(address(protocolVersions), "ProtocolVersions");
+
+        // Set up L2. There are currently no proxies set in the L2 initialization.
```

Comment:
WDYM here?
Should this be a TODO comment?

Reply:
I will add a comment here but we would like to use https://github.com/foundry-rs/foundry/pull/6101 to set up the L2 state so that the same code path used in prod to create the L2 genesis state is used to populate the state that is tested against in the solidity tests